Barotrauma/BarotraumaClient/Source/Screens/CharacterEditorScreen.cs

```diff
@@ -2377,20 +2377,40 @@ private void ValidateJoint(LimbJoint limbJoint)
 
         private Limb GetClosestLimbOnRagdoll(Vector2 targetPos, Func<Limb, bool> filter = null)
         {
-            // TODO: optimize?
-            return character.AnimController.Limbs
-                .Where(l => filter == null ? true : filter(l))
-                .OrderBy(l => Vector2.Distance(SimToScreen(l.SimPosition), targetPos))
-                .FirstOrDefault();
+            Limb closestLimb = null;
+            float closestDistance = float.MaxValue;
+            foreach (Limb l in character.AnimController.Limbs) 
+            {
+                if (filter == null ? true : filter(l)) 
+                {
+                    float distance = Vector2.Distance(SimToScreen(l.SimPosition), targetPos);
```

Comment:
Could use DistanceSquared for a tiny extra improvement

Reply:
big extra improvement

Reply:
Yeah, that's a fair point - I was trying to retain the original functions used though.

Reply:
gj